app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaDetailsPresenter.kt

```diff
@@ -487,10 +494,63 @@ class MangaDetailsPresenter(
                 deleteChapters(selectedChapters, false)
             }
             getChapters()
+            if (preferences.autoUpdateTrack("toggle")) {
+                val newLastChapter = chapters.filter { it.read }.minByOrNull { it.source_order }
+                if (oldLastChapter != newLastChapter) {
+                    updateTrackChapterRead(oldLastChapter, newLastChapter)
+                }
+            }
             withContext(Dispatchers.Main) { controller.updateChapters(chapters) }
         }
     }
 
+    /**
+     * Starts the service that updates the last chapter read in sync services. This operation
+     * will run in a background thread and errors are ignored.
+     */
+    private fun updateTrackChapterRead(oldLastChapter: ChapterItem?, newLastChapter: ChapterItem?) {
+        if (!preferences.autoUpdateTrack("toggle")) return
+
+        val oldChapterRead = oldLastChapter?.chapter_number?.toInt() ?: 0
+        val newChapterRead = newLastChapter?.chapter_number?.toInt() ?: 0
+
+        val trackManager = Injekt.get<TrackManager>()
+
+        // We want these to execute even if the presenter is destroyed so launch on GlobalScope
+        job?.cancel()
+        job = launchIO {
```

Comment:
as mentioned it's fine to use presenterscope for this, it's short calls

Reply:
actually, thinking again of the person who will close pages the moment they mark something as read, I makes sense to useon globalscope. but i'd still use something else like
`(controller.activity as AppCompatActivity).lifecycleScope.launchIO`